packages/gestalt/src/Icon/InternalIcon.tsx

```diff
@@ -0,0 +1,185 @@
+import vrIcons from '../icons-vr-theme/index';
+import useInExperiment from '../useInExperiment';
+
+export type IconColor =
+  | 'default'
+  | 'disabled'
+  | 'subtle'
+  | 'success'
+  | 'error'
+  | 'warning'
+  | 'info'
+  | 'recommendation'
+  | 'inverse'
+  | 'shopping'
+  | 'brandPrimary'
+  | 'light'
+  | 'dark';
+
+type IconName = keyof typeof icons | keyof typeof compactIconsVR | keyof typeof compactIconsClassic;
+type Props = {
+  accessibilityLabel: string;
+  color?: IconColor;
+  dataTestId?: string;
+  icon?: IconName;
+  dangerouslySetSvgPath?: {
+    __path: string;
+  };
+  inline?: boolean;
+  size?: number | string;
+};
+
+// @ts-expect-error - TS2322 - Type 'string[]' is not assignable to type 'readonly ("replace" | "search" | "link" | "text" | "dash" | "3D" | "3D-move" | "360" | "accessibility" | "ad" | "ad-group" | "add" | "add-circle" | "add-layout" | "add-pin" | "add-section" | ... 317 more ... | "wave")[]'.
+const IconNames: ReadonlyArray<IconName> = Object.keys(icons);
+const swapOnRtlIconNames: ReadonlyArray<IconName> = ['list-numbered'];
+
+const flipOnRtlIconNames = [
+  'ads-stats',
+  'ads-overview',
+  'arrow-back',
+  'arrow-circle-forward',
+  'arrow-end',
+  'arrow-forward',
+  'arrow-start',
+  'arrow-up-right',
+  'compose',
+  'chevron-small-left',
+  'chevron-small-right',
+  'chevron-left-circle',
+  'chevron-right-circle',
+  'directional-arrow-left',
+  'directional-arrow-right',
+  'flipVertical',
+  'hand-pointing',
+  'link',
+  'mute',
+  'reorder-images',
+  'send',
+  'sound',
+  'speech',
+  'speech-ellipsis',
+  'switch-account',
+  'text-size',
+  'visit',
+];
+
+/**
+ * [Icons](https://gestalt.pinterest.systems/web/icon) are the symbolic representation of an action or information, providing visual context and improving usability.
+ *
+ * See the [Iconography and SVG guidelines](https://gestalt.pinterest.systems/foundations/iconography/library) to explore the full icon library.
+ *
+ * ![Icon light mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/Icon-list.spec.ts-snapshots/Icon-list-chromium-darwin.png)
+ * ![Icon dark mode](https://raw.githubusercontent.com/pinterest/gestalt/master/playwright/visual-test/Icon-list-dark.spec.ts-snapshots/Icon-list-dark-chromium-darwin.png)
+ *
+ */
+function InternalIcon({
+  accessibilityLabel,
+  color = 'subtle',
+  dangerouslySetSvgPath,
+  dataTestId,
+  icon,
+  inline = false,
+  size = 16,
+}: Props) {
+  const cs = classnames(
+    // @ts-expect-error - TS2345 - Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
+    flipOnRtlIconNames.includes(icon) && styles.rtlSupport,
+    styles[color],
+    styles.icon,
+    { [styles.iconBlock]: !inline },
+  );
+  const isInExperiment = useInExperiment({
+    webExperimentName: 'web_gestalt_visualRefresh',
+    mwebExperimentName: 'web_gestalt_visualRefresh',
+  });
+
+  const getIconPath = (iconToUse?: IconName) => {
```

Comment:
i think we can simplify a lot this logic. 

we can add a size='compact' | 'default' prop and set the viewBox to the right size